hawkular-inventory-impl-tinkerpop-parent/hawkular-inventory-impl-tinkerpop/src/main/java/org/hawkular/inventory/impl/tinkerpop/TinkerpopBackend.java

```diff
@@ -178,7 +179,7 @@ public Element find(CanonicalPath element) throws ElementNotFoundException {
                     });
         }
 
-        return new Page<>(q2.toList(), pager, q.getCount("total"));
+        return new Page<>(q2, pager, q.getCount("total"));
```

Comment:
Same here. The `toList()` was there so that `q.getCount()` reports the correct number, IMHO.